### PR TITLE
Fix #17 - on first import there event doesn't exist yet.

### LINKED ISSUE #17: AssertionError: assert($form_display instanceof EntityFormDisplayInterface) line 64 localgov_events.module
Just a quick note that I'm seeing this on install:

```
AssertionError: assert($form_display instanceof EntityFormDisplayInterface) in assert() (line 64 of /app/web/modules/contrib/localgov_events/localgov_events.module) #0 /app/web/modules/contrib/localgov_events/localgov_events.module(64): assert(false, 'assert($form_di...')
#1 /app/web/modules/contrib/localgov_events/localgov_events.install(18): localgov_events_optional_fields_settings(true, true)
#2 [internal function]: localgov_events_install(true)
#3 /app/web/core/lib/Drupal/Core/Extension/ModuleHandler.php(392): call_user_func_array('localgov_events...', Array)
```

Looks like these fail assertion as they don't exist yet.

### PR DESCRIPTION
**Pull request by ekes**
This is when the module is being installed by a configuration import
that has got the fields attached anyway (or not as the case is).

It shouldn't effect the behaviour, tested, of the installation when
enabling the module for the first time.

### CODE REVIEW

**Comment by finnlewis** [comment]
This looks trivial enough to me and fixes the install problem.
I'll leave it a day or so though in case anyone else wants to review.

**Comment by finnlewis** [comment]
Once this is approved, I think we'd like to tag another release to assist with our deploy on Thursday.

**Comment by andybroomfield** [comment]
Not sure what I'm reviewing @ekes @finnlewis. Is it installing by reimport config or using config installer?

**Comment by andybroomfield** [comment]
Since I can't replicate the issuem, i'd be hesitant about simply returning if the form_display or view_display are empty, as that sounds important.

**Review by andybroomfield — CHANGES_REQUESTED** [review]
I worked out that this error occurs because directories needs to be enabled in the sql dump before trying to import from config. To help avoid confusion in future, please add these steps in steps to reporduce when requesting review.

I'd like to see a test that demostrates the failing behaviour that this fix address before approving.

I'm still not sure that skiping if the displays or form is empty is the right fix here?

So this `function localgov_events_optional_fields_settings` is run when either localgov_directories_page or localgov_directories_venue is installed? And then adds the activity provider and venue fields.

Looking at the docs for [hook_modules_installed](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Extension%21module.api.php/function/hook_modules_installed/8.9.x) there is an extra parameter `bool $is_syncing` which looks like the supported way to check that a configuration import is running, and that could be used to skip over the field install in localgov_events_optional_fields_settings (which a config import should bring in anyway).
Same should apply to hook_install.

**Comment by finnlewis** [comment]
Thanks for looking at this @andybroomfield. 

You are right to point out that directories needs to be installed. On Lambeth we had directories installed first, then tried to enable events, generating the error. 

Just testing again locally but so far I am also finding it hard to replicate the error on #17 AssertionError: assert($form_display instanceof EntityFormDisplayInterface) line 64 localgov_events.module #17

@ekes do you have a qay to reliably reproduce the error?

**Comment by ekes** [comment]
Assuming it's still the case to replicate an assertion error you will have to have assertion errors on. Generally they are in debug environments only - which is why they are nice highlighting a failed assumption
https://www.php.net/manual/en/function.assert.php
https://www.php.net/manual/en/function.assert-options.php

But then if it wasn't failing at the assertion it should be operating on null here
https://github.com/localgovdrupal/localgov_events/pull/18/files#R107
and or here 
https://github.com/localgovdrupal/localgov_events/pull/18/files#R121
Which generally also throws an exception these days. So I assume this will require work to check what the sequence was.

**Comment by finnlewis** [comment]
@andybroomfield @ekes any thoughts on whether we want to keep this one alive ?

**Comment by ekes** [comment]
Yes, if it was an issue we should review this pattern - that we use quite often. Although the implications of not reacting on is_sync'ing presently hurts my head.

**Comment by finnlewis** [comment]
Just discussing this in the Tech Drop-in. Agreed to try to replicate the issue before confirming if this change fixes it.

**Comment by andybroomfield** [comment]
Closed in favour of #73

**Pull request closed without merging**
